tests/performance-test/performance-test.sh

```diff
@@ -49,91 +49,170 @@ ellipse(){
-            }
-        }
-EOF
-)
+    if ! echo $datasources | grep -q "SAFPrometheus"; then
+        echo "Error: unable to find Grafana datasource SAFPrometheus"
+        exit 1
+    fi
+
+    if ! echo $datasources | grep -q "SAFElasticsearch"; then
+        echo "Error: unable to find Grafana datasource SAFElasticsearch"
+        exit 1
+    fi
 }
 
+# Post dashboards to grafana - overwrites if they already exist
 post_dashboards(){
-    echo "Creating new dashboards in Grafana"
+    printf "\n*** Creating new dashboards in Grafana ***\n"
     curl --silent --output /dev/null -d "{\"overwrite\": true, \"dashboard\": $(cat ./grafana/perftest-dashboard.json)}" \
         -H 'Content-Type: application/json' "$GRAF_HOST/api/dashboards/db"
         
     curl --silent --output /dev/null -d "{\"overwrite\": true, \"dashboard\": $(cat ./grafana/prom2-dashboard.json)}" \
         -H 'Content-Type: application/json' "$GRAF_HOST/api/dashboards/db"
 }
 
-# Execute
+# Delete recorded events in Elastic Search
+delete_es_events(){
+    local esPod=$(oc get elasticsearch elasticsearch -o jsonpath='{.status.pods.master.ready[0]}')
+    local secretPath="/etc/elasticsearch/secret/"
+    local esQuery=$(cat <<EOF
+        {
+            "query" : {
+                "bool": {
+                    "must": {
+                        "match_phrase": {
+                            "labels.instance": {
+                                "query": "saf-perftest-notify-*"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+EOF
+)
+    printf "\n*** Deleting ES Events ***\n"
+
+    local response=$(oc exec "$esPod" -- curl -k --silent --trace --cacert "${secretPath}admin-ca" --cert \
+        "${secretPath}admin-cert" --key "${secretPath}admin-key" -X POST "https://localhost:9200/collectd_interface_if/_delete_by_query" \
+        -d "$esQuery")
+}
+
+# Get number of events recorded by elastic search since last clear
+get_es_event_count(){
+        local esPod=$(oc get elasticsearch elasticsearch -o jsonpath='{.status.pods.master.ready[0]}')
+        local secretPath="/etc/elasticsearch/secret/"
+        local esQuery=$(cat <<EOF
+        {
+            "query" : {
+                "bool": {
+                    "must": {
+                        "match_phrase": {
+                            "labels.instance": {
+                                "query": "saf-perftest-notify-*"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+EOF
+)
+    ES_EVENT_RECV_COUNT=$(oc exec "$esPod" -- curl -k --silent --trace --cacert "${secretPath}admin-ca" --cert \
+         "${secretPath}admin-cert" --key "${secretPath}admin-key" -X GET "https://localhost:9200/collectd_interface_if/_count" \
+         -d "$esQuery")
+
+    ES_EVENT_RECV_COUNT=$(echo "$ES_EVENT_RECV_COUNT" | python -c "import sys, json; print json.load(sys.stdin)['count']")
```

Comment:
Bringing back this python dependency I see. We managed to drop it previously when we stopped using performance-test-tg.sh. I won't harp on it since I know you wanted to re-write the whole thing in python anyways ;)

Reply:
I think I gave up after the fourth time trying to parse JSON with regular expressions